### PR TITLE
Refactor MetricsIsolatedST to be more robust

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

This PR refactor MetricsIsolatedST to deploy resources inside the tests inside the big before all. It also merge together several test cases and make the tests more stable.

As part of it, I also added javadoc that describe the steps.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/templates/crd/KafkaTemplates.java [hunk 1]
```diff
@@ -31,6 +31,10 @@ public class KafkaTemplates {
 
     private KafkaTemplates() {}
 
+    private static final String KAFKA_METRICS_CONFIG_REF_KEY = "kafka-metrics-config.yml";
+    private static final String ZOOKEEPER_METRICS_CONFIG_REF_KEY = "zookeeper-metrics-config.yml";
+
+
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/templates/crd/KafkaTemplates.java:37]: Extra line

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java [hunk 2]
```diff
@@ -437,6 +447,185 @@ void testCreateTopicAfterUnsupportedOperation(ExtensionContext extensionContext)
...
+        assertKafkaTopicStatus(topicName, namespace, Ready, 6);
+
+        // Check failed reconciliations
+        // This is currently doesn't work properly and will be changed in UTO implementation - https://github.com/strimzi/proposals/pull/76
+        // assertMetricValueHigherThan(toMetricsCollector, "strimzi_reconciliations_failed_total{kind=\"" + KafkaTopic.RESOURCE_KIND + ",}", 3);
+    }
+
+    @ParallelTest
+    @KRaftNotSupported("TopicOperator is not supported by KRaft mode and is used in this test class")
+    // The test should be merged with testKafkaTopicDifferentStates when the issue wull be fixed
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:583]:
```suggestion
    // The test should be merged with testKafkaTopicDifferentStates when the issue will be fixed
```

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:583]: Out of curiosity, did you try that it still doesn't work? IIRC we had a conversation around that. If not, we can test it afterwards

**Review by im-konge — APPROVED** [review]
LGTM, thanks :)

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java (hunk 2 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:583]: This one is connected to the issue which isn't resolved and I think we discussed that this particular issue still exists. I can try it on Monday if it works.

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java]: Testes are working fine, removing disabled annotation

**Review by scholzj — APPROVED** [review]

**Pull request merged**